lib/util/check-token.ts

```diff
@@ -0,0 +1,63 @@
+import { PlatformId } from '../constants';
+import { logger } from '../logger';
+import { GithubReleasesDatasource } from '../modules/datasource/github-releases';
+import { GithubTagsDatasource } from '../modules/datasource/github-tags';
+import type { PackageFile } from '../modules/manager/types';
+import * as memCache from '../util/cache/memory';
+import * as hostRules from './host-rules';
+import { parseUrl } from './url';
+
+function isGithubUrl(url: string | null | undefined): boolean {
+  const parsedUrl = parseUrl(url);
+  if (!parsedUrl) {
+    return false;
+  }
+
+  const { hostname } = parsedUrl;
+  return hostname.endsWith('.github.com') || hostname === 'github.com';
+}
+
+export function checkGithubToken(
+  packageFiles: Record<string, PackageFile[]>
+): void {
+  const { token } = hostRules.find({
+    hostType: PlatformId.Github,
+    url: 'https://api.github.com',
+  });
+
+  if (token) {
+    logger.trace('GitHub token is found');
+    return;
+  }
+
+  const githubDeps: string[] = [];
+  for (const files of Object.values(packageFiles)) {
+    for (const file of files) {
```

Comment:
```suggestion
    for (const file of files ?? []) {
```
I think files can be null

Reply:
☝️

Reply:
Values of `Record<string, PackageFile[]>` will never be nullable, isn't it?